### PR TITLE
[FLINK-36635][runtime] Remove PhysicalSlotRequestBulkImpl as it is not used

### PR DESCRIPTION
**Pull request by eemario**
## What is the purpose of the change
[FLINK-36635]
This pull request removes PhysicalSlotRequestBulkImpl which is not used anywhere.

## Brief change log
- Remove the class PhysicalSlotRequestBulkImpl
- Modify TestingPhysicalSlotRequestBulkBuilder and PhysicalSlotRequestBulkCheckerImplTest to use SharingPhysicalSlotRequestBulk instead
- Modify the visibility of ExecutionSlotSharingGroup and SharingPhysicalSlotRequestBulk for the tests

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

**Comment by eemario** [comment]
@flinkbot run azure

File: flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/ExecutionSlotSharingGroup.java [hunk 1]
```diff
@@ -31,18 +31,19 @@
...
 /** Represents execution vertices that will run the same shared slot. */
-class ExecutionSlotSharingGroup {
+public class ExecutionSlotSharingGroup {
 
     private final Set<ExecutionVertexID> executionVertexIds;
 
     @Nonnull private final SlotSharingGroup slotSharingGroup;
 
-    ExecutionSlotSharingGroup(@Nonnull SlotSharingGroup slotSharingGroup) {
+    @VisibleForTesting
```
Comment (JunRuiLee) [inline flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/ExecutionSlotSharingGroup.java:40]: VisibleForTesting declares that a function, field, constructor, or entire type is only visible for testing purposes. Therefore, there is no need to add this annotation here.

File: flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SharingPhysicalSlotRequestBulk.java [hunk 2]
```diff
@@ -57,7 +58,8 @@ class SharingPhysicalSlotRequestBulk implements PhysicalSlotRequestBulk {
 
     private final BiConsumer<ExecutionVertexID, Throwable> logicalSlotRequestCanceller;
 
-    SharingPhysicalSlotRequestBulk(
+    @VisibleForTesting
```
Comment (JunRuiLee) [inline flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SharingPhysicalSlotRequestBulk.java:61]: ditto

File: flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/TestingPhysicalSlotRequestBulkBuilder.java [hunk 3]
```diff
@@ -19,35 +19,43 @@
...
 import java.util.Map;
 import java.util.function.BiConsumer;
 
 class TestingPhysicalSlotRequestBulkBuilder {
-    private static final BiConsumer<SlotRequestId, Throwable> EMPTY_CANCELLER = (r, t) -> {};
-    private Map<SlotRequestId, ResourceProfile> pendingRequests = new HashMap<>();
-    private BiConsumer<SlotRequestId, Throwable> canceller = EMPTY_CANCELLER;
+    private static final BiConsumer<ExecutionVertexID, Throwable> EMPTY_CANCELLER = (r, t) -> {};
+    private Map<ExecutionSlotSharingGroup, List<ExecutionVertexID>> executions = new HashMap<>();
+    private Map<ExecutionSlotSharingGroup, ResourceProfile> pendingRequests = new HashMap<>();
```
Comment (JunRuiLee) [inline flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/TestingPhysicalSlotRequestBulkBuilder.java:35]: executions and pendingRequests can be final.

**Comment by eemario** [comment]
@flinkbot run azure

**Review by JunRuiLee — APPROVED** [review]
LGTM.

**Pull request merged**